sdk/search/search-documents/test/internal/serialization.spec.ts

```diff
@@ -49,7 +71,29 @@ describe("serialization.deserialize", () => {
   it("circular", () => {
     const circluarInput: any = { a: null };
     circluarInput.a = circluarInput;
-    assert.throws(() => deserialize(circluarInput));
+    const result = deserialize(circluarInput);
+    assert.deepEqual(circluarInput, result);
+  });
+
+  it("recursive 1", () => {
+    const child = { hello: "world" };
+    const documents = [
+      { id: "1", children: [child] },
+      { id: "2", children: [child] }
+    ];
+    const result = deserialize(documents);
+    assert.deepEqual(documents, result);
+  });
+
+  it("recursive 2", () => {
+    const child = { hello: Infinity, world: -Infinity, universe: NaN };
```

Comment:
this should be going the other way with the server sending the string version, yeah?

```suggestion
    const child = { hello: "INF", world: "-INF", universe: "NaN"};
```